src/kontrol/foundry.py

```diff
@@ -58,9 +58,12 @@ def __init__(
         bug_report: BugReport | None = None,
     ) -> None:
         self._root = foundry_root
-        with (foundry_root / 'foundry.toml').open('rb') as f:
-            self._toml = tomlkit.load(f)
-        self._bug_report = bug_report
+        try:
+            with (foundry_root / 'foundry.toml').open('rb') as f:
+                self._toml = tomlkit.load(f)
+        except FileNotFoundError:
+            _LOGGER.error('No Foundry project found. Exiting.')
+            sys.exit(1)
```

Comment:
This exception should be handled by the command line layer. Otherwise, this would force a library user to handle `SystemExit`.

Reply:
@tothtamas28 Then in this particular case I should surround with a try except the foundry-list call in 

```python
def exec_list(foundry_root: Path, **kwargs: Any) -> None:
    stats = foundry_list(foundry_root=foundry_root)
    print('\n'.join(stats))
```

But this should apply for all `exec` options of kontrol like `prove` and `build`.

Reply:
This indicates that refactoring is necessary before introducing this change.

* Methods like `foundry_list` should take `foundry: Foundry`.
* You can add a helper in `__cli__` that attempts to load a `Foundry` and exists with an error message if `foundry.toml` is not found.

Reply:
Done! Thank you for the suggestion!